examples/rhat_monitor.cpp

```diff
@@ -0,0 +1,367 @@
+// clang++ -std=c++20 -O3 -pthread rhat_monitor.cpp -o rhat_monitor
+// ./rhat_monitor
+
+#include <atomic>
+#include <chrono>
+#include <cmath>
+#include <fstream>
+#include <functional>
+#include <iostream>
+#include <latch>
+#include <numeric>
+#include <random>
+#include <stop_token>
+#include <string>
+#include <thread>
+#include <vector>
+
+#define VERY_INLINE [[gnu::always_inline]] inline
```

Comment:
It is annoying to write, but generally its good to check if the attribute exists before making the macro.
```suggestion
#if defined __has_attribute
#if __has_attribute(always_inline)
#ifndef WALNUTS_STRONG_INLINE
#define WALNUTS_STRONG_INLINE [[gnu::always_inline]] inline
#endif
#endif
#endif
```